lib/routes/route_stats.ex

```diff
@@ -115,6 +116,12 @@ defmodule Routes.RouteStats do
     |> round_to_1_place()
   end
 
+  def median_schedule_adherence(stats_by_route, route_id) when is_binary(route_id) do
```

Comment:
suggestion: Each of the new cases you've added to this file could probably get a quick test case to demonstrate the usage and prevent any regressions.